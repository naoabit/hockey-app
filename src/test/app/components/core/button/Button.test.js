import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '@app/components/core/button/ButtonComponent';

jest.mock('react-redux-i18n');

describe('<Button />', () => {
    describe('enabled', () => {
        const spy = jest.fn();
        const props = {
            className: 'la classe',
            handleClick: spy,
            icon: 'icon',
            text: 'A button',
        };

        const wrapper = shallow(
            <Button {...props}/>,
        );

        it('should render component', () => {
            expect(toJson(wrapper)).toMatchSnapshot();
            
            const text = wrapper.childAt(0);
            const icon = wrapper.childAt(1);
            
            expect(wrapper.props().className).toEqual('la classe button');
            expect(!!wrapper.props().onClick).toBe(true);
            expect(text.is('Translate')).toBe(true);
            expect(icon.text()).toEqual('icon');
        });

        it('should call props.handleClick', () => {
            wrapper.simulate('click', { preventDefault: jest.fn() });
            expect(spy).toHaveBeenCalledTimes(1);
        });
    });
});
