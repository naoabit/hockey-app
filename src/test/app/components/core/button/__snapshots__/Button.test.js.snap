// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> enabled should render component 1`] = `
<div
  className="la classe button"
  onClick={[Function]}
>
  <Translate
    dangerousHTML={true}
    tag="span"
    value="A button"
  />
  <div
    className="icon"
  >
    icon
  </div>
</div>
`;
